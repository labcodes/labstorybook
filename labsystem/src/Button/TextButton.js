import React from "react";
import PropTypes from "prop-types";
import AbstractButton from "./AbstractButton";

export default class TextButton extends React.Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    skin: PropTypes.oneOf(["", "light", "dark"]),
    icon: PropTypes.string,
    size: PropTypes.oneOf(["normal", "small", "large"]),
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    fullWidth: PropTypes.bool,
  };

  static defaultProps = {
    skin: undefined,
    icon: undefined,
    size: "normal",
    disabled: false,
    onClick: undefined,
    fullWidth: false,
  };

  render() {
    const { text, skin, icon, size, disabled, onClick, fullWidth } = this.props;
    return (
      <AbstractButton
        variant="text"
        text={text}
        skin={skin}
        icon={icon}
        size={size}
        disabled={disabled}
        onClick={onClick}
        fullWidth={fullWidth}
      />
    );
  }
}
