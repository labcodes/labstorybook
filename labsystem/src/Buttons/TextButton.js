import React from "react";
import PropTypes from "prop-types";
import AbstractButton from "./AbstractButton";

export default class TextButton extends React.Component {
  static propTypes = {
    /** Text that will be rendered inside the Button. */
    text: PropTypes.string.isRequired,
    /** Skin of the the rendered Button. */
    skin: PropTypes.oneOf(["light", "dark"]),
    /** Type of the icon to be rendered. Won't render an icon if not passed to the component. */
    icon: PropTypes.string,
    /** Size of the Button. */
    size: PropTypes.oneOf(["normal", "small", "large"]),
    /** Disables the Button, cancelling the onClick handler. */
    disabled: PropTypes.bool,
    /** Callback to be executed when the Button is clicked. */
    onClick: PropTypes.func,
    /** Makes the Button occupy its container's full width. */
    fullWidth: PropTypes.bool,
  };

  static defaultProps = {
    skin: undefined,
    icon: undefined,
    size: "normal",
    disabled: false,
    onClick: undefined,
    fullWidth: false,
  };

  render() {
    const { text, skin, icon, size, disabled, onClick, fullWidth } = this.props;
    return (
      <AbstractButton
        variant="text"
        text={text}
        skin={skin}
        icon={icon}
        size={size}
        disabled={disabled}
        onClick={onClick}
        fullWidth={fullWidth}
      />
    );
  }
}
