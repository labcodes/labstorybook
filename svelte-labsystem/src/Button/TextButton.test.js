import { render } from "@testing-library/svelte";
import TextButton from "./TextButton.svelte";

describe("TextButton", () => {
  it("renders with base props", async () => {
    const { container } = render(TextButton, { text: "test" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing disabled as true", async () => {
    const { container, getByText } = render(TextButton, {
      text: "test",
      disabled: true,
    });

    expect(container).toMatchSnapshot();
    expect(getByText("test").disabled).toBeTruthy();
  });

  it("renders as expected when passing a warning skin", async () => {
    const { container } = render(TextButton, { text: "test", skin: "warning" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing a small size", async () => {
    const { container } = render(TextButton, { text: "test", size: "small" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing a wallet icon", async () => {
    const { container } = render(TextButton, { text: "test", icon: "wallet" });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing fullWidth prop", async () => {
    const { container } = render(TextButton, { text: "test", fullWidth: true });
    expect(container).toMatchSnapshot();
  });
});
