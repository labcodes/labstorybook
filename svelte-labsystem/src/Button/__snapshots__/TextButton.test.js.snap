// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextButton renders as expected when passing a small size 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--text lab-btn--small"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`TextButton renders as expected when passing a wallet icon 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--text lab-btn--normal"
    >
      <span
        class="lab-icon lab-icon--wallet lab-icon--white lab-icon--petit lab-btn__icon"
      />
       
      test
    </button>
  </div>
</body>
`;

exports[`TextButton renders as expected when passing a warning skin 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--text lab-btn--normal lab-btn--warning"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`TextButton renders as expected when passing disabled as true 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--text lab-btn--normal"
      disabled=""
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`TextButton renders as expected when passing fullWidth prop 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--text lab-btn--normal lab-btn--block"
    >
       
      test
    </button>
  </div>
</body>
`;

exports[`TextButton renders with base props 1`] = `
<body>
  <div>
    <button
      class="lab-btn lab-btn--text lab-btn--normal"
    >
       
      test
    </button>
  </div>
</body>
`;
